google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/flexible/AppEngineFlexibleFrameworkDetector.java

```diff
@@ -66,8 +69,45 @@ public FileType getFileType() {
   @NotNull
   @Override
   public ElementPattern<FileContent> createSuitableFilePattern() {
-    return StandardPatterns.or(
-        FileContentPattern.fileContent().withName("app.yaml"),
-        FileContentPattern.fileContent().withName("app.yml"));
+    return new AppEngineFlexPattern().withAppEngineFlexYamlContent();
+  }
+
+  /**
+   * IntelliJ API pattern class that checks for App Engine Flex project file presence and checks it
+   * has required configuration lines to avoid spurious detection.
+   */
+  private static final class AppEngineFlexPattern
+      extends ObjectPattern<FileContent, AppEngineFlexPattern> {
+
+    private AppEngineFlexPattern() {
+      super(FileContent.class);
+    }
+
+    private AppEngineFlexPattern withAppEngineFlexYamlContent() {
+      return with(
+          new PatternCondition<FileContent>("with-appengine-java-flexible") {
+            // app engine flex file names.
+            private final Collection<String> APP_ENGINE_FLEX_CONFIG_FILES =
+                ImmutableList.of("app.yaml", "app.yml");
+
+            private static final String APP_ENGINE_FLEX_REQUIRED_YAML_CONTENT = "runtime:";
+
+            @Override
+            public boolean accepts(@NotNull FileContent fileContent, ProcessingContext context) {
+              // checks for flex engine file names and then checks for required configuration line
+              // inside.
+              boolean nameMatch = APP_ENGINE_FLEX_CONFIG_FILES.contains(fileContent.getFileName());
+              if (nameMatch) {
+                Scanner scanner = new Scanner(fileContent.getContentAsText().toString());
+                while (scanner.hasNextLine()) {
+                  if (scanner.nextLine().startsWith(APP_ENGINE_FLEX_REQUIRED_YAML_CONTENT)) {
```

Comment:
should you trim the line before matching with the required content? I know yaml is indentation sensitive but I'm wondering if its still possible for it to be valid with leading spaces on the "runtime:" line.

Reply:
Yes, I thought it's supposed to be on the first position only, but if it's not and it can be indented, I will trim the line.

Reply:
I guess since its a top-level element it shouldn't be trimmed, on second thought

Reply:
it doesn't have to be in the first position, but it is a top-level element

Reply:
Ok, then we leave it as it is, correct? Otherwise we'll be checking on nested element

Reply:
yea i think its good as is